src/Workspaces/Core/Portable/Serialization/SerializableSourceText.cs

```diff
@@ -65,15 +66,19 @@ public async ValueTask<SourceText> GetTextAsync(CancellationToken cancellationTo
             return await Storage!.ReadTextAsync(cancellationToken).ConfigureAwait(false);
         }
 
-        public static async ValueTask<SerializableSourceText> FromTextDocumentStateAsync(TextDocumentState state, CancellationToken cancellationToken)
+        public static ValueTask<SerializableSourceText> FromTextDocumentStateAsync(TextDocumentState state, CancellationToken cancellationToken)
         {
             if (state.Storage is ITemporaryTextStorageWithName storage)
             {
-                return new SerializableSourceText(storage);
+                return new ValueTask<SerializableSourceText>(new SerializableSourceText(storage));
             }
             else
             {
-                return new SerializableSourceText(await state.GetTextAsync(cancellationToken).ConfigureAwait(false));
+                return SpecializedTasks.TransformWithoutIntermediateCancellationExceptionAsync(
```

Comment:
i think it woudl be good to have a comment here.

Reply:
My intent is to have the method name eliminate the need for a separate comment. Open to suggestions for improving upon the current name.